### ISSUE TITLE
Application constantly creating/exiting threads, bogomips (ApplicationInsightsDiagnostics.json)?

### ISSUE THREAD

**Issue by alystair** [webprofusion/certify#624]
Context: Trying out tooling for a single domain, telemetry is turned off... and UI is totally closed.

I noticed the service.exe is never idling - wasting many cycles. The service is constantly creating/terminating a thread and attempting to access 'ApplicationInsightsDiagnostics.json' which as far as I can see doesn't exist.

![image](https://user-images.githubusercontent.com/159109/199218832-06cf6a72-fab0-4851-ba22-7da64f5958ae.png)

This can be observed using procmon / procexp tools from [Sysinternals](https://learn.microsoft.com/en-ca/sysinternals/), a set of Microsoft tools.

It's harder to justify using this SSL manager if it's wasting resources.

**Comment by webprofusion-chrisc** [comment]
Thanks, I've definitely not seen that before, have you restarted the service since turning off telemetry? The reason I ask is the way our app is designed it will just have an null telemetry object so it won't even be able to access telemetry at all if it's disabled at startup.

**Comment by webprofusion-chrisc** [comment]
Is outgoing https disabled via your firewall? The behavior obviously suggests it's trying to do something and failing. If the problem persists please also append the contents of C:\ProgramData\Certify\appsettings.json

**Comment by alystair** [comment]
You're absolutely correct, after restarting the service I'm now only seeing the thread creation/exit :)

![image](https://user-images.githubusercontent.com/159109/199313891-106dc614-4806-4401-ab75-cdcba581e6b6.png)

```json
{
  "SettingsSchemaVersion": 1,
  "CheckForUpdatesAtStartup": true,
  "EnableAppTelematics": false,
  "IgnoreStoppedSites": true,
  "EnableValidationProxyAPI": true,
  "EnableEFS": false,
  "EnableDNSValidationChecks": false,
  "RenewalIntervalDays": 30,
  "RenewalIntervalMode": "DaysAfterLastRenewal",
  "MaxRenewalRequests": 0,
  "EnableHttpChallengeServer": true,
  "LegacySettingsUpgraded": true,
  "IsInstanceRegistered": false,
  "InstanceId": "9a61df16-5cae-46e2-871c-56469e165934",
  "Language": null,
  "EnableCertificateCleanup": true,
  "EnableStatusReporting": true,
  "CertificateCleanupMode": 1,
  "DefaultCertificateAuthority": "letsencrypt.org",
  "CertificateAuthorityFallback": null,
  "DefaultKeyCredentials": null,
  "EnableAutomaticCAFailover": false,
  "IncludeExternalPlugins": false,
  "FeatureFlags": null,
  "NtpServer": "pool.ntp.org",
  "DefaultCertificateStore": null,
  "EnableExternalCertManagers": true
}
```

**Comment by alystair** [comment]
I think the most efficient way would be to use native windows scheduler and only activate the service around expiry dates? This way ~80mb RAM is also freed up :)

**Comment by webprofusion-chrisc** [comment]
Thanks, we definitely still need to look at this issue as there is probably more we can do. 

We want to try to trim the service down a little. We support a bunch of features that in turn have dependencies which are loaded at runtime (some due to .net reflection as part of plugin loads) so we do want to get the service smaller. The UI does need the service in order to operate as the UI itself does not do any of the real work.

If you're looking for a particularly lightweight option then you could consider Posh-ACME (PowerShell), win-acme or even certbot. Certify The Web does quite a lot more than some people need.

**Comment by webprofusion-chrisc** [comment]
As a little bit of background to our service based approach, some users are actively managing up to 18,000 certs on a single server, so the service is frequently checking for work to do, however it does that hourly so I still need to check what else is creating this thread every 10 mins. I suspect its our log files being flushed periodically.

**Comment by alystair** [comment]
Thanks, I'll look into Posh-ACME for my use case.

**Comment by alystair** [comment]
It's actually every few seconds, not every 10 minutes. Hope this helps :)

![image](https://user-images.githubusercontent.com/159109/200148362-378cf4ef-2baa-44ab-a18b-667213b97cb5.png)
![image](https://user-images.githubusercontent.com/159109/200148402-e4323fd3-84a8-4e4a-8932-bc9c6b292a1b.png)

**Comment by webprofusion-chrisc** [comment]
Application Insights removed in v7.
